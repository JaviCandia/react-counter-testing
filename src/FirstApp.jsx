import PropTypes from "prop-types";

const nameList = () => {
  const names = ["Javier", "Maribel", "Chofis", "Wero"];
  return names.map((name, index) => <li key={index}> {name} </li>);
};

export const FirstApp = ({ title, subtitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <h2 data-testid="test-subtitle"> {subtitle} </h2>
      <p>{name}</p>
      <p>Checa esta lista:</p>
      <ul>{nameList()}</ul>
    </>
  );
};

FirstApp.propTypes = {
  subtitle: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  name: "Fernando",
  subtitle: "No hay subtitulo",
  // title: "No hay titulo",
};
