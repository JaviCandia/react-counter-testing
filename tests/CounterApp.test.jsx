import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe("CounterApp", () => {
  const initialValue = 100;

  it("should match snapshot", () => {
    const { container } = render(<CounterApp />);

    expect(container).toMatchSnapshot();
  });

  it("should show initial value", () => {
    render(<CounterApp value={initialValue} />);

    expect(screen.getByText(initialValue)).toBeTruthy();
  });

  it("should increment initial value", () => {
    render(<CounterApp value={initialValue} />);

    fireEvent.click(screen.getByRole('button', {name: 'btn-add'}));

    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(`${initialValue + 1}`);
  });

  it("should decrement initial value", () => {
    render(<CounterApp value={initialValue} />);

    fireEvent.click(screen.getByRole('button', {name: 'btn-substract'}));

    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(`${initialValue - 1}`);
  });

  it("should reset to initial value", () => {
    render(<CounterApp value={initialValue} />);

    fireEvent.click(screen.getByRole('button', {name: 'btn-add'}));
    fireEvent.click(screen.getByRole('button', {name: 'btn-reset'}));

    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(`${initialValue}`);
  });
});
