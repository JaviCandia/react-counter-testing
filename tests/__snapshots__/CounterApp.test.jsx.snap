// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp should match snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    0
     
  </h2>
  <button
    aria-label="btn-substract"
  >
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
  <button
    aria-label="btn-add"
  >
    +1
  </button>
</div>
`;
